server/sonar-db-dao/src/main/java/org/sonar/db/DefaultDatabase.java

```diff
@@ -0,0 +1,189 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import javax.sql.DataSource;
+import org.apache.commons.dbcp.BasicDataSource;
+import org.apache.commons.dbcp.BasicDataSourceFactory;
+import org.apache.commons.dbutils.DbUtils;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.config.Settings;
+import org.sonar.api.database.DatabaseProperties;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.dialect.Dialect;
+import org.sonar.db.dialect.DialectUtils;
+import org.sonar.db.profiling.NullConnectionInterceptor;
+import org.sonar.db.profiling.ProfiledConnectionInterceptor;
+import org.sonar.db.profiling.ProfiledDataSource;
+import org.sonar.process.logging.LogbackHelper;
+
+import static java.lang.String.format;
+
+/**
+ * @since 2.12
+ */
+public class DefaultDatabase implements Database {
+
+  private static final Logger LOG = Loggers.get(Database.class);
+
+  private static final String DEFAULT_URL = "jdbc:h2:tcp://localhost/sonar";
+  private static final String SONAR_JDBC = "sonar.jdbc.";
+  private static final String SONAR_JDBC_DIALECT = "sonar.jdbc.dialect";
+  private static final String SONAR_JDBC_URL = "sonar.jdbc.url";
+
+  private final LogbackHelper logbackHelper;
+  private final Settings settings;
+  private ProfiledDataSource datasource;
+  private Dialect dialect;
+  private Properties properties;
+
+  public DefaultDatabase(LogbackHelper logbackHelper, Settings settings) {
+    this.logbackHelper = logbackHelper;
+    this.settings = settings;
+  }
+
+  @Override
+  public void start() {
+    try {
+      initSettings();
+      initDataSource();
+      checkConnection();
+
+    } catch (Exception e) {
+      throw new IllegalStateException("Fail to connect to database", e);
+    }
+  }
+
+  @VisibleForTesting
+  void initSettings() {
+    properties = new Properties();
+    completeProperties(settings, properties, SONAR_JDBC);
+    completeDefaultProperty(properties, DatabaseProperties.PROP_URL, DEFAULT_URL);
+    doCompleteProperties(properties);
+
+    dialect = DialectUtils.find(properties.getProperty(SONAR_JDBC_DIALECT), properties.getProperty(SONAR_JDBC_URL));
+    properties.setProperty(DatabaseProperties.PROP_DRIVER, dialect.getDefaultDriverClassName());
+  }
+
+  private void initDataSource() throws Exception {
+    // but it's correctly caught by start()
+    LOG.info("Create JDBC data source for {}", properties.getProperty(DatabaseProperties.PROP_URL, DEFAULT_URL));
+    BasicDataSource basicDataSource = (BasicDataSource) BasicDataSourceFactory.createDataSource(extractCommonsDbcpProperties(properties));
+    datasource = new ProfiledDataSource(basicDataSource, NullConnectionInterceptor.INSTANCE);
+    datasource.setConnectionInitSqls(dialect.getConnectionInitStatements());
+    datasource.setValidationQuery(dialect.getValidationQuery());
+    enableSqlLogging(datasource, logbackHelper.getLoggerLevel("sql") == Level.TRACE);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)